fatal error: Python.h: No such file or directory

I have anaconda3 installed with python 3 where I created a new conda env to install simpleitk.
After editing the meta.yml to have the latest master version (version: "your_custom_version", git_tag: master), I tried to do `conda build recipe` in the conda env, but it gives the error (below). There are other packages working fine in the same env.

```
In file included from /home/bk17/anaconda3/conda-bld/simpleitk_1500393985191/work/Wrapping/Python/sitkPyCommand.cxx:22:0:
/home/bk17/anaconda3/conda-bld/simpleitk_1500393985191/work/Wrapping/Python/sitkPython.h:71:20: fatal error: Python.h: No such file or directory
 #include <Python.h>
                    ^
compilation terminated.
In file included from /home/bk17/anaconda3/conda-bld/simpleitk_1500393985191/work/build/SimpleITK-build/Wrapping/Python/SimpleITKPYTHON_wrap.cxx:11:0:
/home/bk17/anaconda3/conda-bld/simpleitk_1500393985191/work/Wrapping/Python/sitkPython.h:71:20: fatal error: Python.h: No such file or directory
 #include <Python.h>
                    ^
compilation terminated.
make[5]: *** [Wrapping/Python/CMakeFiles/SimpleITK_PYTHON.dir/sitkPyCommand.cxx.o] Error 1
make[5]: *** Waiting for unfinished jobs....
make[5]: *** [Wrapping/Python/CMakeFiles/SimpleITK_PYTHON.dir/SimpleITKPYTHON_wrap.cxx.o] Error 1
make[4]: *** [Wrapping/Python/CMakeFiles/SimpleITK_PYTHON.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [SimpleITK-prefix/src/SimpleITK-stamp/SimpleITK-build] Error 2
make[1]: *** [CMakeFiles/SimpleITK.dir/all] Error 2
make: *** [all] Error 2
Traceback (most recent call last):
  File "/home/bk17/anaconda3/bin/conda-build", line 6, in <module>
    sys.exit(conda_build.cli.main_build.main())
  File "/home/bk17/anaconda3/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 342, in main
    execute(sys.argv[1:])
  File "/home/bk17/anaconda3/lib/python3.6/site-packages/conda_build/cli/main_build.py", line 333, in execute
    noverify=args.no_verify)
  File "/home/bk17/anaconda3/lib/python3.6/site-packages/conda_build/api.py", line 183, in build
    need_source_download=need_source_download, config=config, variants=variants)
  File "/home/bk17/anaconda3/lib/python3.6/site-packages/conda_build/build.py", line 1615, in build_tree
    built_packages=built_packages,
  File "/home/bk17/anaconda3/lib/python3.6/site-packages/conda_build/build.py", line 1072, in build
    utils.check_call_env(cmd, env=env, cwd=src_dir)
  File "/home/bk17/anaconda3/lib/python3.6/site-packages/conda_build/utils.py", line 670, in check_call_env
    return _func_defaulting_env_to_os_environ(subprocess.check_call, *popenargs, **kwargs)
  File "/home/bk17/anaconda3/lib/python3.6/site-packages/conda_build/utils.py", line 666, in _func_defaulting_env_to_os_environ
    return func(_args, **kwargs)
  File "/home/bk17/anaconda3/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/bin/bash', '-x', '-e', '/home/bk17/anaconda3/conda-bld/simpleitk_1500393985191/work/conda_build.sh']' returned non-zero exit status 2.
```

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -8,7 +8,7 @@ else
 fi
 
 # When building 32-bits on 64-bit system this flags is not automatically set by conda-build
-if [ $ARCH == 32 -a "${OSX_ARC:-notosx}" == "notosx" ]; then
+if [ $ARCH == 32 -a "${OSX_ARCH:-notosx}" == "notosx" ]; then
     export CFLAGS="${CFLAGS} -m32"
     export CXXFLAGS="${CXXFLAGS} -m32"
 fi

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -1,10 +1,7 @@
 #!/bin/bash
 
 if [ `uname` == Darwin ]; then
-    SO_EXT='dylib'
     CMAKE_ARGS="-D CMAKE_OSX_DEPLOYMENT_TARGET:STRING=${MACOSX_DEPLOYMENT_TARGET}"
-else
-    SO_EXT='so'
 fi
 
 # When building 32-bits on 64-bit system this flags is not automatically set by conda-build
@@ -17,10 +14,10 @@ BUILD_DIR=${SRC_DIR}/build
 mkdir ${BUILD_DIR}
 cd ${BUILD_DIR}
 
-MY_PY_VER=${PY_VER}
-if [ $PY3K -eq "1" ]; then
-    MY_PY_VER="${MY_PY_VER}m"
-fi
+
+PYTHON_INCLUDE_DIR=$(${PYTHON} -c 'import sysconfig;print("{0}".format(sysconfig.get_path("platinclude")))')
+PYTHON_LIBRARY_DIR=$(${PYTHON} -c 'import sysconfig;print("{0}/{1}".format(*map(sysconfig.get_config_var, ("LIBDIR", "LDLIBRARY"))))')
+
 
 cmake \
     -D "CMAKE_CXX_FLAGS:STRING=-fvisibility=hidden -fvisibility-inlines-hidden ${CXXFLAGS}" \
@@ -42,8 +39,8 @@ cmake \
     -D ITK_USE_SYSTEM_ZLIB:BOOL=ON \
     -D "CMAKE_SYSTEM_PREFIX_PATH:FILEPATH=${PREFIX}" \
     -D "PYTHON_EXECUTABLE:FILEPATH=${PYTHON}" \
-    -D "PYTHON_INCLUDE_DIR:PATH=$PREFIX/include/python${MY_PY_VER}" \
-    -D "PYTHON_LIBRARY:FILEPATH=$PREFIX/lib/libpython${MY_PY_VER}.${SO_EXT}" \
+    -D "PYTHON_INCLUDE_DIR:PATH=${PYTHON_INCLUDE_DIR}" \
+    -D "PYTHON_LIBRARY_DIR:PATH=${PYTHON_LIBRARY_DIR}" \
     -D "SWIG_EXECUTABLE:FILEPATH=${PREFIX}/bin/swig" \
     "${SRC_DIR}/SuperBuild"
 

